src/nus/cs2103/addressbook/AddressBook.java

```diff
@@ -30,188 +39,258 @@
-        while (!isExitRequested) {
-            latestUserInput = getUserInput();
-            String feedback = parseAndExecuteCommand(latestUserInput);
+        loadDataFromStorage();
+        while (true) {
+            userCommand = getUserInput();
+            echoUserCommand(userCommand);
+            String feedback = executeCommand(userCommand);
             showToUser(feedback);
         }
-        cleanup();
+    }
+
+    /*
+     * ==============NOTE TO STUDENTS======================================
+     * The method header comment can be omitted if the method is trivial
+     * and the header comment is going to be almost identical to the method
+     * signature anyway.
+     * ====================================================================
+     */
+    private static void showWelcomeMessage() {
+        showToUser(MESSAGE_WELCOME);
+    }
+
+    /*
+     * ==============NOTE TO STUDENTS======================================
+     * Parameter description can be omitted from the method header comment
+     * if the parameter name is self-explanatory.
+     * In the method below, '@param userInput' comment has been omitted.
+     * ====================================================================
+     */
+    /**
+     * Echoes the user input back to the user.
+     */
+    private static void echoUserCommand(String userCommand) {
+        showToUser(System.lineSeparator() + "[Command entered:" + userCommand + "]");
     }
 
     /*
      * ==============NOTE TO STUDENTS==========================================
-     * If the reader wants a deeper understanding of the solution, he/she can go
+     * If the reader wants a deeper understanding of the solution, she can go
      * to the next level of abstraction by reading the methods (given below)
      * that is referenced by the method above.
      * ====================================================================
      */
 
     /**
      * Processes the program main method run arguments.
-     * Also handles parsing and validating the arguments.
-     * If any arguments are invalid, the corresponding default configuration is used.
+     * If a valid storage file is specified, sets up that file for storage.
+     * Otherwise sets up the default file for storage.
      * 
-     * @param args full program run arguments passed to application main method
+     * @param args full program arguments passed to application main method
      */
     private static void processProgramArgs(String[] args) {
-        informUserIfProgramArgsAreInvalid(args);
-        storageFilePath = args.length == 1 ? args[0] : DEFAULT_STORAGE_FILEPATH;
+        if (args.length >= 2) {
+            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+            exitProgram();
+        }
+
+        if (args.length == 1) {
+            setupGivenFileForStorage(args[0]);
+        }
+
+        if(args.length == 0) {
+            setupDefaultFileForStorage();
+        }
+
     }
```

Comment:
Rejigged this method to make it more cohesive and SLAP. Found a but in the process.